src/Umbraco.Web.BackOffice/Filters/UmbracoAuthorizeFilter.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.AspNetCore.Routing;
+using System;
+using Umbraco.Core;
+using Umbraco.Extensions;
+using Umbraco.Web.Security;
+
+namespace Umbraco.Web.BackOffice.Filters
+{
+
+    /// <summary>
+    /// Ensures authorization is successful for a back office user.
+    /// </summary>
+    public class UmbracoAuthorizeFilter : IAuthorizationFilter
+    {
+        private readonly bool _requireApproval;
+
+        /// <summary>
+        /// Can be used by unit tests to enable/disable this filter
+        /// </summary>
+        internal static bool Enable = true;
+        
+        private readonly IUmbracoContextAccessor _umbracoContext;
+        private readonly IRuntimeState _runtimeState;
+        private readonly LinkGenerator _linkGenerator;
+        private readonly bool _redirectToUmbracoLogin;
+        private string _redirectUrl;
+
+        /// <summary>
+        /// Default constructor
+        /// </summary>
+        /// <param name="umbracoContext"></param>
+        /// <param name="runtimeState"></param>
+        /// <param name="linkGenerator"></param>
+        public UmbracoAuthorizeFilter(
+            IUmbracoContextAccessor umbracoContext, IRuntimeState runtimeState, LinkGenerator linkGenerator)
+        {
+            _umbracoContext = umbracoContext ?? throw new ArgumentNullException(nameof(umbracoContext));
+            _runtimeState = runtimeState ?? throw new ArgumentNullException(nameof(runtimeState));
+            _linkGenerator = linkGenerator ?? throw new ArgumentNullException(nameof(linkGenerator));
+        }
+
+        /// <summary>
+        /// Constructor with redirect umbraco login behavior
+        /// </summary>
+        /// <param name="umbracoContext"></param>
+        /// <param name="runtimeState"></param>
+        /// <param name="linkGenerator"></param>
+        /// <param name="redirectToUmbracoLogin">If true will redirect to the umbraco login page if not authorized</param>
+        public UmbracoAuthorizeFilter(
+            IUmbracoContextAccessor umbracoContext, IRuntimeState runtimeState, LinkGenerator linkGenerator,
+            bool redirectToUmbracoLogin)
+            : this(umbracoContext, runtimeState, linkGenerator)
+        {            
+            _redirectToUmbracoLogin = redirectToUmbracoLogin;
+        }
+
+        /// <summary>
+        /// Constructor with redirect url behavior
+        /// </summary>
+        /// <param name="umbracoContext"></param>
+        /// <param name="runtimeState"></param>
+        /// /// <param name="linkGenerator"></param>
+        /// <param name="redirectUrl">If specified will redirect to this URL if not authorized</param>
+        public UmbracoAuthorizeFilter(
+            IUmbracoContextAccessor umbracoContext, IRuntimeState runtimeState, LinkGenerator linkGenerator,
+            string redirectUrl)
+            : this(umbracoContext, runtimeState, linkGenerator)
+        {
+            _redirectUrl = redirectUrl;
```

Comment:
I'm not a fan of multiple constructors setting different fields.. I think we should make a private construtor handling all fields, and use that for each of them..

So something like this:

```
   privateUmbracoAuthorizeFilter(
            IUmbracoContextAccessor umbracoContext, IRuntimeState runtimeState, LinkGenerator linkGenerator,
 bool redirectToUmbracoLogin, string redirectUrl)
        {
            _umbracoContext = umbracoContext ?? throw new ArgumentNullException(nameof(umbracoContext));
            _runtimeState = runtimeState ?? throw new ArgumentNullException(nameof(runtimeState));
            _linkGenerator = linkGenerator ?? throw new ArgumentNullException(nameof(linkGenerator));
            _redirectToUmbracoLogin = redirectToUmbracoLogin;
            _redirectUrl = redirectUrl;
        }
```

Having multiple constructors setting fields are doomed to go wrong at some time